File: src/data/models/Account.ts
import type { AccountType } from "./AccountType"
import type { Currency } from "./Currencies"

export default interface Account {
    id: number
    name: string
    type: AccountType
    currency: Currency
    startingBalance: number
    createdAt: Date
    updatedAt: Date
    archivedAt?: Date
}

File: API/FinancePlanner.Domain/Models/Account.cs
namespace FinancePlanner.Domain.Models;

public class Account
{
    public long Id { get; set; }
    public string Name { get; set; } = null!;
    public AccountType Type { get; set; }
    public Currency Currency { get; set; }
    public decimal StartingBalance { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? ArchivedAt { get; set; }
    public string UserId { get; set; }
    public ApplicationUser User { get; set; }
}